feat: Revamp the website with Jekyll

zusyed opened the pull request:
This PR updates the website to use Jekyll templates.

This addresses https://github.com/tophat/opensource.tophat.com/issues/64

# QA

- You need Ruby to [install Jekyll](https://jekyllrb.com/docs/). I had trouble installing Jekyll with the builtin system ruby so I installed RVM by following [this guide](https://usabilityetc.com/articles/ruby-on-mac-os-x-with-rvm/) but YMMV.
- Once Jekyll is installed, simply run the `bundle exec jekyll serve` command to get the site up and running.
- Verify the site looks as expected.
- Add or remove a project from the [projects.yaml](https://github.com/tophat/opensource.tophat.com/blob/convert-to-jekyll/_data/projects.yml) file (we can use JSON files instead of YAML files but I think YAML looks cleaner. More info [here](https://jekyllrb.com/docs/datafiles/)) and verify the projects show up as expected.

### Enabling Slack invite forms
- The Slack invite sections are also converted but they are disabled since Slack invite flow is broken at the moment. To enable the invite slack form at the bottom of the index page, uncomment [this](https://github.com/tophat/opensource.tophat.com/blob/convert-to-jekyll/_includes/join.html#L18-L24).
- To enable the slack invite page, remove `slack.html` from the [config.yml](https://github.com/tophat/opensource.tophat.com/blob/convert-to-jekyll/_config.yml#L5) file. Then restart the server (re-run `bundle exec jekyll serve`) for the changes to take effect.

ju-nc commented:
This looks great!

I believe we want to include a `Gemfile`, right? 

Also, when I QA'd, I noticed the project github links say "GitHub", but at the top the link text is "Github"

zusyed commented:
Yes, `Gemfile` is recommended by [Jekyll](https://jekyllrb.com/docs/step-by-step/10-deployment/). Gemfile is now added.

Also updated the links at the top to say "GitHub".

ju-nc commented on README.md:
```diff
@@ -11,6 +11,10 @@ Top Hats open source portal
 
+### Local Development
+This site uses Jekyll
+ - Install Jekyll by following docs on [their website](https://jekyllrb.com/docs/)
```
nb:

could this be updated to run "bundle install"?

zusyed replied:
Yes, you can use Bundler to install/update Jekyll. Here is the documentation they have for it: https://jekyllrb.com/tutorials/using-jekyll-with-bundler/

I followed the quickstart guide to install Jekyll which is why the quickstart link is in the README. I can add the link to install using bundler too so people are aware that is also an option.

zusyed replied:
Updated the README to include that link. Unfortunately, that removed your approval. Can you approve again?

ju-nc reviewed (approved):
CR and QA LGTM! One non blocking comment

mcataford commented on Gemfile:
```diff
@@ -0,0 +1,30 @@
+source "https://rubygems.org"
```
**[nb]**

IMO we can take out the boilerplatey comments here.

zusyed replied:
I agree. Updated to remove those comments.

zusyed replied:
@mcataford Can you approve the PR if you think this PR is good to merge? The change remove the approval.

mcataford replied:
Done, sorry for the delay, in a meeting. 😅

ju-nc reviewed (approved):
LGTM

mcataford reviewed (commented):
Looks neat to me!

One thing re:setup/bootstrapping -- any chance we can have a bootstrap script that would set up what we need to build locally? It can be done as a follow-up since it's more devtooley.

zusyed commented:
> Looks neat to me!
> 
> One thing re:setup/bootstrapping -- any chance we can have a bootstrap script that would set up what we need to build locally? It can be done as a follow-up since it's more devtooley.

I can take a look at that. I also think it makes sense to do it as a follow-up.
